stl/inc/xutility

```diff
@@ -5504,9 +5504,75 @@ _NODISCARD constexpr _Iter_diff_t<_InIt> _Count_pr(_InIt _First, const _InIt _La
     return _Count;
 }
 
+enum class _TrimResult : unsigned char { _Continue, _Break, _ReturnFalse, _ReturnTrue };
+
+template <class _BidIt1, class _BidIt2, class _Pr>
+_NODISCARD _CONSTEXPR20 _TrimResult _Trim_equal(
+    _BidIt1& _First1, _BidIt1& _Last1, _BidIt2& _First2, _BidIt2& _Last2, _Pr _Pred) {
+    _STL_INTERNAL_CHECK(_First1 != _Last1);
+    _STL_INTERNAL_CHECK(_STD distance(_First1, _Last1) == _STD distance(_First2, _Last2));
+    if (_Pred(*_First1, *_First2)) {
+        do {
+            ++_First1;
+            ++_First2;
+            if (_First1 == _Last1) {
+                // only one element is left
+                return _Pred(*_First1, *_First2) ? _TrimResult::_ReturnTrue : _TrimResult::_ReturnFalse;

```

Comment:
This is dereferencing `_Last1` which seems wrong

Reply:
`_Last1` and `_Last2` are iterators just before end.

Reply:
Then we would need to give them a different name. I *think* the convention is to use `_Final` or something

I think there is only one occurence in  `inplace_merge` where we have `_Last` not pointing beyond the end and even there I hate it with all my heart.

Reply:
AFAIK `_Final` means an iterator corresponding to `_Last` (if it is a sentinel) https://github.com/microsoft/STL/pull/1772#discussion_r599784133. What do you think of name `_Back`?

Reply:
`_Back` sounds good, I wish we would have used something like `_SLast` for the sentinel and then `_Last` for the corresponding iterator.